sdk/storage/storage-blob/src/generateSASUrl.ts

```diff
@@ -0,0 +1,22 @@
+import { BlobClient, GenerateSASUrlOptions } from "./BlobClient";
+import { SharedKeyCredential } from "./credentials/SharedKeyCredential";
+import { isNode } from "@azure/core-http";
+import { generateBlobSASQueryParameters } from "./BlobSASSignatureValues";
+
+BlobClient.prototype.generateSASUrl = function(
+  options: GenerateSASUrlOptions,
```

Comment:
I believe we don't want to expose full set of SAS token.  So should only take `'r'` permission and this would be `generatePublicReadOnlyUrl` or similar.